File: services/api/src/atlas_api/models/workspace.py
"""Workspace domain models — mirrors packages/shared/src/types/workspace.ts."""

from pydantic import BaseModel, ConfigDict

from atlas_api.models.enums import WorkspaceRole


class WorkspaceMember(BaseModel):
    model_config = ConfigDict(frozen=True)

    user_id: str
    role: WorkspaceRole
    joined_at: str


class Workspace(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    slug: str
    name: str
    description: str | None
    owner_id: str
    members: tuple[WorkspaceMember, ...]
    is_private: bool
    created_at: str
    updated_at: str

File: packages/shared/src/types/workspace.ts
import type { WorkspaceId, UserId } from "./ids.js";
import type { WorkspaceRole } from "./enums.js";

export interface WorkspaceMember {
  readonly userId: UserId;
  readonly role: WorkspaceRole;
  readonly joinedAt: string;
}

export interface Workspace {
  readonly id: WorkspaceId;
  readonly slug: string;
  readonly name: string;
  readonly description: string | null;
  readonly ownerId: UserId;
  readonly members: readonly WorkspaceMember[];
  readonly isPrivate: boolean;
  readonly createdAt: string;
  readonly updatedAt: string;
}
